doc/source/dev_guide/custom_reader.rst

```diff
@@ -500,6 +500,11 @@ needs to implement a few methods:
 
 On top of that, two attributes need to be defined: ``start_time`` and
 ``end_time``, that define the start and end times of the sensing.
+See the :ref:`time_metadata` section for a description of the different
+times that Satpy readers typically use and what times should be used
+for the ``start_time`` and ``end_time``. If these times are not part of
+the DataArray metadata returned by ``get_dataset`` it will be added from
+these attributes.
```

Comment:
Maybe it's worth specifying here that `start_time` and `end_time` are not only being added to the dataset `attrs` if missing, but they will also be overwritten if already present. See https://github.com/pytroll/satpy/blob/8aa4280be8ea2b27565fd077a4f67f05c883036b/satpy/readers/yaml_reader.py#L726-L727
CC @strandgren as we found this out lately together.

Reply:
Good point!

Reply:
Ok I've just pushed a clarification for this. Thanks. And if you or @strandgren want to make a PR to have start/end_time set only if they don't already exist I think I'd be OK with that. I'm not sure I can think of a use case where we *need* to overwrite these values. Overall, as you've noticed, it is surprising to the reader developer.